decimal.go

```diff
@@ -1102,6 +1104,8 @@ func (z *Big) SetString(s string) (*Big, bool) {
 				return nil, false
 			}
 			err = nerr.Err
+		} else if z.compact == 0 {
```

Comment:
shouldn't this just be an "if"? since z.compact == 0 is possible even if err == nil